mqtt/mqtt-broker/src/error.rs

```diff
@@ -9,6 +9,18 @@ pub struct Error {
 
 #[derive(Debug, Fail, PartialEq)]
 pub enum ErrorKind {
+    #[fail(display = "Authentication: failed to execute.")]
+    AuthExecution,
```

Comment:
It's better to move this as a inner error for Auth error.
`Error::Authenticate(AuthenticateError::Execution(...))`